src/test/java/org/openmrs/contrib/qaframework/helper/Page.java

```diff
@@ -0,0 +1,408 @@
+
+			@Override
+			public Boolean apply(WebDriver driver) {
+				return hasFocus(tag, attr, value);
+			}
+		});
+	}
+
+	public void waitForJsVariable(final String varName) {
+		waiter.until(new ExpectedCondition<Boolean>() {
+
+			@Override
+			public Boolean apply(WebDriver driver) {
+				return ((JavascriptExecutor) driver)
+						.executeScript("return (typeof " + varName
+								+ "  !== 'undefined') ? " + varName + " : null") != null;
+			}
+		});
+	}
+
+	public void waitForElementToBeHidden(By by) {
+		waiter.until(ExpectedConditions.invisibilityOfElementLocated(by));
+	}
+
+	public void waitForElementToBeEnabled(By by) {
+		waiter.until(ExpectedConditions.elementToBeClickable(by));
+	}
+
+	public void acceptAlert() {
+		waiter.until(ExpectedConditions.alertIsPresent());
+		Alert alert = driver.switchTo().alert();
+		alert.accept();
+	}
+
+	boolean hasFocus(String id) {
+		return (Boolean) ((JavascriptExecutor) driver).executeScript(
+				"return jQuery('#" + id + "').is(':focus')", new Object[]{});
+	}
+
+	boolean hasFocus(String tag, String attr, String value) {
+		return (Boolean) ((JavascriptExecutor) driver).executeScript(
+				"return jQuery('" + tag + "[" + attr + "=" + value
+						+ "]').is(':focus')", new Object[]{});
+	}
+
+	public void waitForElement(By by) {
+		waiter.until(ExpectedConditions.visibilityOfElementLocated(by));
+	}
+
+	public void waitForElementWithSpecifiedMaxTimeout(By by, long secs) {
+		WebDriverWait waiter = new WebDriverWait(driver, secs);
+		waiter.until(pageReady);
+		waiter.until(ExpectedConditions.visibilityOfElementLocated(by));
+	}
+
+	public void waitForTextToBePresentInElement(By by, String text) {
+		waiter.until(ExpectedConditions.textToBePresentInElementLocated(by,
+				text));
+	}
+
+	public boolean containsTextWithoutWait(String text) {
+		return driver.getPageSource().contains(text);
+	}
+
+	public Boolean containsText(String text) {
+		return driver.getPageSource().contains(text);
+	}
+
+	public List<String> getValidationErrors() {
+		List<String> validationErrors = new ArrayList<String>();
+		for (WebElement webElement : driver.findElements(By
+				.className("field-error"))) {
+			if (StringUtils.isNotBlank(webElement.getText())) {
+				validationErrors.add(webElement.getText());
+			}
+		}
+		for (WebElement webElement : driver.findElements(By.className("error"))) {
+			if (StringUtils.isNotBlank(webElement.getText())) {
+				validationErrors.add(webElement.getText());
+			}
+		}
+		return validationErrors;
+	}
+
+	public String queryJsForAttribute(String cssHandle, String attribute) {
+		return (String) ((JavascriptExecutor) driver).executeScript(String
+				.format("return document.querySelector('%s').%s", cssHandle,
+						attribute));
+	}
+
+	public void setAttributeWithJs(String cssHandle, String attribute,
+			String value) {
+		((JavascriptExecutor) driver).executeScript(String.format(
+				"document.querySelector('%s').%s = '%s'", cssHandle, attribute,
+				value));
+	}
+}
```

Comment:
Please include a blank line at the end of files (you can usually configure your IDE to do this for you)

```suggestion
}

```

Reply:
Resolved thanks